jdaviz/core/tests/test_region_translators.py

```diff
@@ -345,3 +350,91 @@ def test_translation_polygon():
+    assert isinstance(reg, PolygonSkyRegion)
+    assert_allclose(reg.vertices.ra.degree, [10, 30, 50])
+    assert_allclose(reg.vertices.dec.degree, [20, 40, 60])
+    # Check that the default frame is ICRS
+    assert isinstance(reg.vertices.frame, ICRS)
+    # Check that the default unit is deg
+    assert reg.vertices.ra.unit == u.deg
+
+
+def test___create_circle_skyregion_from_coords():
+    coords = [10.0, 20.0, 30.0]
+    reg = _create_circle_skyregion_from_coords(coords)
+    assert isinstance(reg, CircleSkyRegion)
+    assert_allclose(reg.center.ra.degree, 10)
+    assert_allclose(reg.center.dec.degree, 20)
+    # Check that the default frame is ICRS
+    assert isinstance(reg.center.frame, ICRS)
+    # Check that the default unit is deg
+    assert reg.center.ra.unit == u.deg
+
+
+def test___create_ellipse_skyregion_from_coords():
+    coords = [10.0, 20.0, 30.0, 40.0, 50.0]
+    reg = _create_ellipse_skyregion_from_coords(coords)
+    assert isinstance(reg, EllipseSkyRegion)
+    assert_allclose(reg.center.ra.degree, 10)
+    assert_allclose(reg.center.dec.degree, 20)
+    # Check that the default frame is ICRS
+    assert isinstance(reg.center.frame, ICRS)
+    # Check that the default unit is deg
+    assert reg.center.ra.unit == u.deg
+
+
+@pytest.mark.parametrize('stcs_string, expected', [
+    ('POLYGON ICRS 10.0 20.0', True),
+    ('CIRCLE ICRS 10.0 20.0', True),
+    ('ELLIPSE ICRS 10.0 20.0', True),
+    ('polygon icrs 10.0 20.0', True),
+    ('circle icrs 10.0 20.0', True),
+    ('ellipse icrs 10.0 20.0', True),
+    ('POLYGON J2000 10.0 20.0', True),
+    ('CIRCLE J2000 10.0 20.0', True),
+    ('ELLIPSE J2000 10.0 20.0', True),
+    ('POLYGON FK5 10.0 20.0', True),
+    ('CIRCLE FK5 10.0 20.0', True),
+    ('ELLIPSE FK5 10.0 20.0', True),
+    ('POLYGON 10.0 20.0', True),
+    ('CIRCLE 10.0 20.0', True),
+    ('ELLIPSE 10.0 20.0', True),
+    ('POLYGON ICRS 1.0', False),
+    ('POLYGON ICRS 1 2 3', False),
+    ('POLYGON 1.0', False),
+    ('POLYGON 1 2 3', False),
+    ('not a STC-S string', False),
+    # Invalid input type (not a string) should return False instead of raising an error
+    (123, False),
+])
+def test_is_stcs_string(stcs_string, expected):
+    assert is_stcs_string(stcs_string) is expected
+
+
+@pytest.mark.parametrize('stcs_string, expected', [
+    ('POLYGON ICRS 10.0 20.0 30.0 40.0 50.0 60.0',
+     PolygonSkyRegion(vertices=SkyCoord(ra=[10, 30, 50], dec=[20, 40, 60], unit='deg'))),
+    ('CIRCLE ICRS 10.0 20.0 30.0',
+     CircleSkyRegion(center=SkyCoord(ra=10, dec=20, unit='deg'), radius=30 * u.deg)),
+    ('ELLIPSE ICRS 10.0 20.0 30.0 40.0 50.0',
+     EllipseSkyRegion(center=SkyCoord(ra=10, dec=20, unit='deg'),
+                      width=30 * u.deg, height=40 * u.deg, angle=50 * u.deg)),
+])
+def test_stcs_string2region_correctness(stcs_string, expected):
+    reg = stcs_string2region(stcs_string)
+    assert isinstance(reg, expected.__class__)
+    assert reg == expected

```

Comment:
If `reg == expected`, shouldn't we expect the test above to always pass? I think the latter test is sufficient.